frappe/public/js/frappe/form/form.js

```diff
@@ -292,13 +292,19 @@ frappe.ui.form.Form = class FrappeForm {
 		for (const d of data) {
 			for (const condition of d) {
 				let [doctype, field, operator, value] = condition;
+				if (value.includes("eval")) {
+					// if condition type of 'like' is used then remove % from value
+					value = value.replace(/%/g, "");
+					// get the value to calculate
+					value = value.split("eval:")[1];
+					value = frappe.utils.eval(value, { doc: this.doc, frappe });
```

Comment:
For child docs here we should pass `parent` as actual doc and `doc` as child table row.